pydicom/fileset.py

```diff
@@ -1508,34 +1508,42 @@ def find_values(
 
         Returns
         -------
-        list of object
-            A list of value(s) for the element available in the instances.
+        list of object(s), or dict of lists of object(s)
+            If single element was queried: A list of value(s) for the element
+                available in the instances.
+            If list of elements was queried: A dict of element value pairs with
+                lists of value(s) for the elements available in the instances.
         """
-        has_element = False
-        results = []
+        single_element = not isinstance(elements, list)
+        if single_element:
+            elements = [elements]
```

Comment:
mypy does not like if the type of an object changes. You can do something like:
```py
element_list = elements if isinstance(elements, list) else [elements]
```
and use `element_list` instead of `elements` in the rest of the method.